From bc97b6c5f6d64f214c1c5f9514db6b483a2709b4 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Thu, 7 Sep 2017 13:57:49 +0800
Subject: [PATCH 3323/5242] MLK-17491-28 clk: imx: clk-pfdv2: add missing
 CLK_SET_RATE_GATE flag

commit  baab3994f5ebd34e25b6b9a0305ddc76ceac2d99 from
https://source.codeaurora.org/external/imx/linux-imx.git

According to reference manual, pfdv2 can't set rate without gating clock.
So we should add CLK_SET_RATE_GATE flag accordingly.

Fixes: 78ef764871d6 ("MLK-13441-5 ARM: imx: add new clk types")
Cc: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-pfdv2.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/imx/clk-pfdv2.c b/drivers/clk/imx/clk-pfdv2.c
index b74f080..141ec611 100644
--- a/drivers/clk/imx/clk-pfdv2.c
+++ b/drivers/clk/imx/clk-pfdv2.c
@@ -158,9 +158,9 @@ struct clk *imx_clk_pfdv2(const char *name, const char *parent_name,
 
 	init.name = name;
 	init.ops = &clk_pfdv2_ops;
-	init.flags = 0;
 	init.parent_names = &parent_name;
 	init.num_parents = 1;
+	init.flags = CLK_SET_RATE_GATE;
 
 	pfd->hw.init = &init;
 
-- 
1.7.9.5

